temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowTaskHandler.java

```diff
@@ -247,27 +246,33 @@ private Result failureToWFTResult(
               .build();
       return new WorkflowTaskHandler.Result(workflowType, response, null, null, null, false);
     }
+
+    if (!shutdownFn.apply()) {
+      WorkflowExecution execution = workflowTask.getWorkflowExecution();
+      log.warn(
+          "Workflow task processing failure. startedEventId={}, WorkflowId={}, RunId={}. If seen continuously the workflow might be stuck.",
+          workflowTask.getStartedEventId(),
+          execution.getWorkflowId(),
+          execution.getRunId(),
+          e);
+    }
+
     // Only fail workflow task on the first attempt, subsequent failures of the same workflow task
     // should timeout. This is to avoid spin on the failed workflow task as the service doesn't
     // yet increase the retry interval.
     if (workflowTask.getAttempt() > 1) {
+      /*
+       *TODO we probably shouldn't swallow Error even if workflowTask.getAttempt() == 1.
```

Comment:
IMO we should only catch `Exception` and not `Throwable` which includes `Error`. The latter signals something is seriously wrong (JVM, disk, mem, etc) and if we really have cases where `Error` is valuable, we should localize the try/catch for them and wrap in an `Exception`. But I understand this is existing behavior.

Reply:
It's not exactly the case in JavaSDK. We have to utilize errors for some SDK functionality. But I agree that we should cleanup how we treat Errors, that's why the comment. Not going to work on it right now though.